File: examples/ruby/sinatra/forward_multiple.rb
require 'sinatra'
require 'builder'

set :bind, '0.0.0.0'
set :port, 3000

post '/' do
  logger.info "event: #{params[:event]}"
  logger.info "call_id: #{params[:call_id]}"
  logger.info "from: #{params[:from]}"
  logger.info "to: #{params[:to]}"

  headers 'Content-Type' => 'application/xml'
  xml = Builder::XmlMarkup.new(indent: 4)
  xml.instruct!

  # Forward to two destinations at the same time
  xml.Response {
    xml.Forward {
      xml.Target {
        xml.Number('022129191999')
        xml.Number('7777abcdefg@fpbx.de')
      }
    }
  }
end

File: examples/node/expressjs/forward_multiple.js
var express = require('express');
var bodyParser = require('body-parser');
var xml = require('xml');

var app = express();
app.use(bodyParser.urlencoded({extended: false}));

app.post('/', function (request, response) {
  var event = request.body.event;
  var call_id = request.body.call_id;
  var from = request.body.from;
  var to = request.body.to;

  console.log('event: ' + event);
  console.log('call_id: ' + call_id);
  console.log('from: ' + from);
  console.log('to: ' + to);

  response.set('Content-Type', 'application/xml');

  // Forward to two destinations at the same time
  response.send(
    xml({
      Response: [
        {
          Forward: [{
            Target: [
              {Number: "022129191999"},
              {Number: "7777abcdefg@fpbx.de"}
            ]
          }]
        }
      ]
    })
  );
});

app.listen(3000);